ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/launcher/TracingBlock.java

```diff
@@ -58,6 +64,62 @@
 
 public class TracingBlock {
 
+	/**
+	 * This class encapsulates all calls to the tracing options manager and runs them using a
+	 */
+	private static final class TracingOptionsManagerDelegate {
+
+		private IRunnableContext fRunnableContext;
+
+		public void setRunnableContext(IRunnableContext runnableContext) {
+			fRunnableContext = runnableContext;
+		}
+
+		Map<String, String> getTracingTemplateCopy() {
+			return runShowingProgress(PDEUIMessages.TracingBlock_initializing_tracing_template,
+					monitor -> PDECore.getDefault().getTracingOptionsManager().getTracingTemplateCopy(monitor));
+		}
+
+		public Map<String, String> getTracingOptions(Map<String, String> options) {
+			return runShowingProgress(PDEUIMessages.TracingBlock_initializing_tracing_options,
+					monitor -> PDECore.getDefault().getTracingOptionsManager().getTracingOptions(options, monitor));
+		}
+
+		public Map<String, String> getTemplateTable(String pluginId) {
+			String taskName = MessageFormat.format(PDEUIMessages.TracingBlock_initializing_tracing_template_table,
+					pluginId);
+			return runShowingProgress(taskName,
+					monitor -> PDECore.getDefault().getTracingOptionsManager().getTemplateTable(pluginId, monitor));
+		}
+
+		private Map<String, String> runShowingProgress(String taskName,
+				Function<IProgressMonitor, Map<String, String>> task) {
+			try {
+				final Map<String, String> result = new HashMap<>();
+
+				// due to a bug
+				// (https://github.com/eclipse-platform/eclipse.platform/issues/769),
+				// the task needs to be forked otherwise the UI
+				// does not show the progress indicator
+				fRunnableContext.run(true, false, monitor -> {
+					SubMonitor subMonitor = SubMonitor.convert(monitor, taskName, 1);
+					result.putAll(task.apply(subMonitor.split(1)));
+				});
```

Comment:
Instead of a `Function` you could use a `BiFunction<TracingOptionsManager, IProgressMonitor, Map<String, String>>` and move the repetitive calls of `PDECore.getDefault().getTracingOptionsManager()` to this method. This would even allow to use a Method-Reference in one caller.
Furthermore if always the same task-name is used, we would save passing that as an argument.
```suggestion
		private Map<String, String> runShowingProgress(
				BiFunction<TracingOptionsManager, IProgressMonitor, Map<String, String>> task) {
			try {
				final Map<String, String> result = new HashMap<>();
				// due to a bug
				// (https://github.com/eclipse-platform/eclipse.platform/issues/769),
				// the task needs to be forked otherwise the UI
				// does not show the progress indicator
				fRunnableContext.run(true, false, monitor -> {
					SubMonitor subMonitor = SubMonitor.convert(monitor,
							PDEUIMessages.TracingBlock_initializing_tracing_options, 1);
					result.putAll(task.apply(PDECore.getDefault().getTracingOptionsManager(), subMonitor.split(1)));
				});
```

Reply:
Done